### ISSUE TITLE
Outdated sniff: WordPress.WP.EnqueuedResourceParameters.NotInFooter

### ISSUE THREAD

**Issue by luistar15** [WordPress/WordPress-Coding-Standards#2420]
## Bug Description
[`$in_footer` is no longer a parameter of `wp_register_script`](https://developer.wordpress.org/reference/functions/wp_register_script/#parameters).
Now is part of `$args` parameter.

**Comment by jrfnl** [comment]
@luistar15 Thanks for bringing this to our attention.

Just had a quick look: this change was made in WP 6.3, so will need a minimum WP based toggle. Additionally, what with `$args` being an array, this will now become much, much harder to detect.

**Comment by dingo-d** [comment]
> Just had a quick look: this change was made in WP 6.3, so will need a minimum WP based toggle. Additionally, what with `$args` being an array, this will now become much, much harder to detect.

Even better, that argument can be a boolean or an array (which was done to avoid BC break). 

It seems that at least the array has specific keys so these can be checked but this definitely complicates things.

**Comment by jrfnl** [comment]
True, the array has specific keys, but what with it being an array, chances of it not being defined in the function call, but elsewhere are much much higher (class constant, class property, passed in via a function parameter, declared within the same function, but not in the function call, conditionally declared)

To be honest, I think it will mean that the sniff will have to be removed as we can no longer reliably detect anything about this.
